File: 0112-path-sum/0112-path-sum.java

class Solution {
    public boolean hasPathSum(TreeNode root, int targetSum) {
        if(root==null) return false;
        int s=targetSum-root.val;
        if(root.left==null && root.right==null){
            if(s==0) return true;
            else  return false;
            }
        boolean l=hasPathSum(root.left,s);
        boolean r=hasPathSum(root.right,s);
        return l|| r;
    }
}

File: 0112-path-sum/0112-path-sum.cpp

class Solution {
public:
    bool hasPathSum(TreeNode* root, int sum) {
        if(root==NULL) return false;
        int s=sum-root->val;
        if(root->left==NULL && root->right==NULL){
            if(s==0) return true;
            else false;
        }
        bool l=hasPathSum(root->left,s);
        bool r=hasPathSum(root->right,s);
        return l || r;
    }
};